File: Linked-List/python/doublelinkedlist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.prev = None
        self.next = None

class DoublyLinkedList:
    def __init__(self):
        self.head = None

    def append(self, value):
        new_node = Node(value)

        if self.head is None:
            self.head = new_node
        else:
            temp = self.head
            while temp.next is not None:
                temp = temp.next
            temp.next = new_node
            new_node.prev = temp

    def log_values(self):
        temp = self.head
        while temp is not None:
            print(temp.data, end=" ")
            temp = temp.next
        print()

# Example usage:
my_list = DoublyLinkedList()
my_list.append(1)
my_list.append(2)
my_list.append(3)

print("Doubly Linked List values:")
my_list.log_values()

File: Linked-List/cpp/doublelinkedlist.cpp
#include <iostream>

using namespace std;

struct Node {
  int data;
  Node *prev;
  Node *next;
};

class DoublyLinkedList {
private:
  Node *head;

public:
  DoublyLinkedList() { head = nullptr; }

  void append(int value) {
    Node *newNode = new Node;
    newNode->data = value;
    newNode->prev = nullptr;
    newNode->next = nullptr;

    if (head == nullptr) {
      head = newNode;
    } else {
      Node *temp = head;
      while (temp->next != nullptr) {
        temp = temp->next;
      }
      temp->next = newNode;
      newNode->prev = temp;
    }
  }

  void logValues() {
    Node *temp = head;
    while (temp != nullptr) {
      cout << temp->data << " ";
      temp = temp->next;
    }
    cout << endl;
  }
};

int main() {
  DoublyLinkedList myList;
  myList.append(1);
  myList.append(2);
  myList.append(3);

  cout << "Doubly Linked List values: ";
  myList.logValues();

  return 0;
}
